### ISSUE TITLE
conda-build 2.1.7 fails when run_test.py is not Python 3 compatible

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda-build#1850]
Trying to build a Python 2 only recipe (which uses Python 2 syntax, such as the `print` statement), fails when trying to test (even though I'm running a conda-build using Python 2 also).   Is the `run_test.py` required to be always written in a Python 2 and 3 compatible way?

**Comment by jakirkham** [comment]
Do you have the recipe available somewhere or perhaps a dummy one that reproduces the problem?

**Comment by ilanschnell** [comment]
The recipe I first ran into this problem with is: https://github.com/ContinuumIO/anaconda-recipes/tree/master/python-2.7
However, as this takes a long time to build, I added a dummy run_test file (containing `print "Hello World!"` to https://github.com/ContinuumIO/anaconda-recipes/tree/master/grin, which is fine as grin only supports Python 2.

**Comment by ilanschnell** [comment]
Ah, I see what is going on now.  Conda-build adds a header to the file which (among other things adds `from __future__ import print_function` to the test script.  So, the following script will fail even when running with Python 2.7:
```
from __future__ import print_function
print "Hello World!"
```

**Comment by ilanschnell** [comment]
So, the solution (as I see it) would be remove:
```
from __future__ import absolute_import, division, print_function
```
from the header, or to tell people that they are not allowed to use the `print` statement in `run_test.py` (which seems overly restrictive to me).

**Comment by msarahan** [comment]
I think this is the commit that broke it: https://github.com/conda/conda-build/commit/fbaf954523a21bf1456d1a5d0f6b02c284ccc6bb

I'll look into it for a 2.1.8 release soon.

**Comment by msarahan** [comment]
fixed by #1876, closing.
